fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/Clock1.scala

```diff
@@ -39,50 +44,119 @@ object Clock1 extends LowPriorityClockInstances {
+    override def epoch: Long = time.toEpochSecond
+    @deprecated("use nowZoned")
+    override def now(accuracy: ClockAccuracy): ZonedDateTime = nowZoned(accuracy)
+
+    override def nowLocal(accuracy: ClockAccuracy, zone: ZoneId): LocalDateTime = ClockAccuracy.applyAccuracy(time.toLocalDateTime, accuracy)
+    override def nowOffset(accuracy: ClockAccuracy, zone: ZoneId): OffsetDateTime = ClockAccuracy.applyAccuracy(time.toOffsetDateTime, accuracy)
+    override def nowZoned(accuracy: ClockAccuracy, zone: ZoneId): Identity[ZonedDateTime] = ClockAccuracy.applyAccuracy(time, accuracy)
+    override def monotonicNano: Long = nano
+  }
+
+  final class ConstantOffset(time: OffsetDateTime, nano: Long) extends Clock1[Identity] {
     override def epoch: Long = time.toEpochSecond
-    override def now(accuracy: ClockAccuracy): ZonedDateTime = ClockAccuracy.applyAccuracy(time, accuracy)
-    override def nowLocal(accuracy: ClockAccuracy): LocalDateTime = now(accuracy).toLocalDateTime
-    override def nowOffset(accuracy: ClockAccuracy): OffsetDateTime = now(accuracy).toOffsetDateTime
+    @deprecated("use nowZoned")
+    override def now(accuracy: ClockAccuracy): ZonedDateTime = nowZoned(accuracy)
+
+    override def nowLocal(accuracy: ClockAccuracy, zone: ZoneId): LocalDateTime = ClockAccuracy.applyAccuracy(time.toLocalDateTime, accuracy)
+    override def nowOffset(accuracy: ClockAccuracy, zone: ZoneId): OffsetDateTime = ClockAccuracy.applyAccuracy(time, accuracy)
+    override def nowZoned(accuracy: ClockAccuracy, zone: ZoneId): Identity[ZonedDateTime] = ClockAccuracy.applyAccuracy(time.toZonedDateTime, accuracy)
     override def monotonicNano: Long = nano
   }
 
   sealed trait ClockAccuracy
   object ClockAccuracy {
+    @deprecated("Use ClockAccuracy.RAW (but better set the limit explicitly!)")
     case object DEFAULT extends ClockAccuracy
+
+    /** The accuracy will not be changed. Generally speaking, it's a bad idea because the actual accuracy might differ between JVM versions (e.g. 11 vs 17)
+      */
+    case object RAW extends ClockAccuracy
+
+    /** Highest precision, although it might be unsafe to use it because some tools (e.g. PostgreSQL) may be implicitly truncating nanosecond-precision timestamps.
+      */
     case object NANO extends ClockAccuracy
     case object MILLIS extends ClockAccuracy
+
+    /**
+      * This is the safest choice for PostgreSQL which is known to truncate timestamps to microseconds
+      */
     case object MICROS extends ClockAccuracy
     case object SECONDS extends ClockAccuracy
     case object MINUTES extends ClockAccuracy
     case object HOURS extends ClockAccuracy
 
-    def applyAccuracy(now: ZonedDateTime, clockAccuracy: ClockAccuracy): ZonedDateTime = {
+    private trait TruncatableTime[T] {
+      def truncatedTo(timestamp: T, unit: TemporalUnit): T
+    }
+    private object TruncatableTime {
+      implicit object TruncatableZoned extends TruncatableTime[ZonedDateTime] {
```

Comment:
implicit object slows down implicit search without much benefit...
```suggestion
      implicit lazy val TruncatableZoned: TruncatableTime[ZonedDateTime]  = new TruncatableTime[ZonedDateTime] {
```